--- SDL_mixer-1.2.5/effect_position.c.bad_code	2004-04-22 14:01:10.540048437 +0200
+++ SDL_mixer-1.2.5/effect_position.c	2004-04-22 14:01:12.165824694 +0200
@@ -153,16 +153,17 @@
 
     for (i = 0; i < len; i += sizeof (Uint32)) {
 #if (SDL_BYTEORDER == SDL_BIG_ENDIAN)
-        *(p++) = (d[l[(*p & 0xFF000000) >> 24]] << 24) |
+        *p = (d[l[(*p & 0xFF000000) >> 24]] << 24) |
                  (d[r[(*p & 0x00FF0000) >> 16]] << 16) |
                  (d[l[(*p & 0x0000FF00) >>  8]] <<  8) |
                  (d[r[(*p & 0x000000FF)      ]]      ) ;
 #else
-        *(p++) = (d[r[(*p & 0xFF000000) >> 24]] << 24) |
+        *p = (d[r[(*p & 0xFF000000) >> 24]] << 24) |
                  (d[l[(*p & 0x00FF0000) >> 16]] << 16) |
                  (d[r[(*p & 0x0000FF00) >>  8]] <<  8) |
                  (d[l[(*p & 0x000000FF)      ]]      ) ;
 #endif
+	p++;
     }
 }
 
@@ -226,16 +227,17 @@
 
     for (i = 0; i < len; i += sizeof (Uint32)) {
 #if (SDL_BYTEORDER == SDL_BIG_ENDIAN)
-        *(p++) = (d[l[((Sint16)(Sint8)((*p & 0xFF000000) >> 24))+128]] << 24) |
+        *p = (d[l[((Sint16)(Sint8)((*p & 0xFF000000) >> 24))+128]] << 24) |
                  (d[r[((Sint16)(Sint8)((*p & 0x00FF0000) >> 16))+128]] << 16) |
                  (d[l[((Sint16)(Sint8)((*p & 0x0000FF00) >>  8))+128]] <<  8) |
                  (d[r[((Sint16)(Sint8)((*p & 0x000000FF)      ))+128]]      ) ;
 #else
-        *(p++) = (d[r[((Sint16)(Sint8)((*p & 0xFF000000) >> 24))+128]] << 24) |
+        *p = (d[r[((Sint16)(Sint8)((*p & 0xFF000000) >> 24))+128]] << 24) |
                  (d[l[((Sint16)(Sint8)((*p & 0x00FF0000) >> 16))+128]] << 16) |
                  (d[r[((Sint16)(Sint8)((*p & 0x0000FF00) >>  8))+128]] <<  8) |
                  (d[l[((Sint16)(Sint8)((*p & 0x000000FF)      ))+128]]      ) ;
 #endif
+	p++;
     }
 
 
