--- SDL_mixer/music.c	2004/11/15 23:34:56	1.58
+++ SDL_mixer/music.c	2005/08/21 20:54:52	1.65
@@ -329,6 +327,9 @@
 int open_music(SDL_AudioSpec *mixer)
 {
 	int music_error;
+#ifdef LIBMIKMOD_MUSIC
+	CHAR *list;
+#endif
 
 	music_error = 0;
 #ifdef WAV_MUSIC
@@ -390,11 +391,17 @@
 	md_mode |= DMODE_HQMIXER|DMODE_SOFT_MUSIC|DMODE_SURROUND;
 #endif
 #ifdef LIBMIKMOD_MUSIC
-	if(!MikMod_InfoDriver())
+	list = MikMod_InfoDriver();
+	if ( list )
+	  free(list);
+	else
 #endif
-	MikMod_RegisterAllDrivers();
+	MikMod_RegisterDriver(&drv_nos);
 #ifdef LIBMIKMOD_MUSIC
-	if(!MikMod_InfoLoader())
+	list = MikMod_InfoLoader();
+	if ( list )
+	  free(list);
+	else
 #endif
 	MikMod_RegisterAllLoaders();
 	if ( MikMod_Init() ) {
