diff -ur SDL_mixer-1.2.7.OLD/playmus.c SDL_mixer-1.2.7/playmus.c
--- SDL_mixer-1.2.7.OLD/playmus.c	2006-05-14 17:08:06.000000000 -0400
+++ SDL_mixer-1.2.7/playmus.c	2006-08-25 15:38:22.000000000 -0400
@@ -65,7 +65,7 @@
 {
 	char buf[10];
 
-	printf("Available commands: (p)ause (r)esume (h)alt > ");
+	printf("Available commands: (p)ause (r)esume (h)alt volume(v#) > ");
 	fflush(stdin);
 	scanf("%s",buf);
 	switch(buf[0]){
@@ -78,6 +78,9 @@
 	case 'h': case 'H':
 		Mix_HaltMusic();
 		break;
+	case 'v': case 'V':
+		Mix_VolumeMusic(atoi(buf+1));
+		break;
 	}
 	printf("Music playing: %s Paused: %s\n", Mix_PlayingMusic() ? "yes" : "no", 
 		   Mix_PausedMusic() ? "yes" : "no");
