diff -ur SDL_mixer-1.2.6.OLD/timidity/config.h SDL_mixer-1.2.6/timidity/config.h
--- SDL_mixer-1.2.6.OLD/timidity/config.h	2004-12-16 00:20:14.000000000 -0500
+++ SDL_mixer-1.2.6/timidity/config.h	2006-04-23 11:28:44.000000000 -0400
@@ -185,14 +185,8 @@
 #define LITTLE_ENDIAN
 #endif
 
-/* DEC MMS has 64 bit long words */
-#ifdef DEC
 typedef unsigned int uint32;
 typedef int int32; 
-#else
-typedef unsigned long uint32;
-typedef long int32; 
-#endif
 typedef unsigned short uint16;
 typedef short int16;
 typedef unsigned char uint8;
