--- ./mikmod/mikmod.h
+++ ./mikmod/mikmod.h
@@ -85,7 +85,8 @@
 
 /*@DOES_NOT_HAVE_SIGNED@*/
 
-#if defined(__alpha)
+#include <stdint.h>
+#if __WORDSIZE == 64
 /* 64 bit architectures */
 
 typedef signed char     SBYTE;      /* 1 byte, signed */
--- ./mikmod/mikmod_internals.h
+++ ./mikmod/mikmod_internals.h
@@ -69,7 +69,8 @@
 /*========== More type definitions */
 
 /* SLONGLONG: 64bit, signed */
-#if defined(__alpha)
+#include <stdint.h>
+#if __WORDSIZE == 64
 typedef long		SLONGLONG;
 #define NATIVE_64BIT_INT
 #elif defined(__WATCOMC__)
